test/forkMainnet/LimitOrderSwap/GroupFill.t.sol

```diff
@@ -0,0 +1,384 @@
+        Snapshot memory maker0MakerToken = BalanceSnapshot.take({ owner: orders[0].maker, token: orders[0].makerToken });
+
+        // order1 35 DAI -> 35 USDT
+        orders[1] = LimitOrder({
+            taker: address(0),
+            maker: makers[1],
+            takerToken: USDT_ADDRESS,
+            takerTokenAmount: 35 * 1e6,
+            makerToken: DAI_ADDRESS,
+            makerTokenAmount: 35 ether,
+            makerTokenPermit: defaultPermit,
+            feeFactor: 0,
+            expiry: defaultExpiry,
+            salt: defaultSalt
+        });
+        makerSigs[1] = _signLimitOrder(makerPrivateKeys[1], orders[1]);
+        makerTokenAmounts[1] = orders[1].makerTokenAmount;
+        Snapshot memory maker1TakerToken = BalanceSnapshot.take({ owner: orders[1].maker, token: orders[1].takerToken });
+        Snapshot memory maker1MakerToken = BalanceSnapshot.take({ owner: orders[1].maker, token: orders[1].makerToken });
+
+        // order2 1000 USDT -> 1000 DAI
+        orders[2] = LimitOrder({
+            taker: address(0),
+            maker: makers[2],
+            takerToken: DAI_ADDRESS,
+            takerTokenAmount: 1000 ether,
+            makerToken: USDT_ADDRESS,
+            makerTokenAmount: 1000 * 1e6,
+            makerTokenPermit: defaultPermit,
+            feeFactor: 0,
+            expiry: defaultExpiry,
+            salt: defaultSalt
+        });
+        makerSigs[2] = _signLimitOrder(makerPrivateKeys[2], orders[2]);
+        makerTokenAmounts[2] = orders[0].takerTokenAmount + orders[1].takerTokenAmount;
+        Snapshot memory maker2TakerToken = BalanceSnapshot.take({ owner: orders[2].maker, token: orders[2].takerToken });
+        Snapshot memory maker2MakerToken = BalanceSnapshot.take({ owner: orders[2].maker, token: orders[2].makerToken });
+
+        address[] memory profitTokens;
+        limitOrderSwap.fillLimitOrderGroup({
+            orders: orders,
+            makerSignatures: makerSigs,
+            makerTokenAmounts: makerTokenAmounts,
+            profitTokens: profitTokens,
+            unwrapAmount: 0
+        });
+
+        // small orders maker should be fully filled
+        maker0TakerToken.assertChange(int256(orders[0].takerTokenAmount));
+        maker0MakerToken.assertChange(-int256(orders[0].makerTokenAmount));
+        maker1TakerToken.assertChange(int256(orders[1].takerTokenAmount));
+        maker1MakerToken.assertChange(-int256(orders[1].makerTokenAmount));
+        // large order maker should gets partial filled
+        maker2TakerToken.assertChange(int256(orders[0].makerTokenAmount + orders[1].makerTokenAmount));
+        maker2MakerToken.assertChange(-int256(makerTokenAmounts[2]));
+        // check order filled amount
+        assertEq(limitOrderSwap.orderHashToMakerTokenFilledAmount(getLimitOrderHash(orders[2])), makerTokenAmounts[2]);
+    }
+
+    function testGroupFillWithWETHUnwrap() public {
```

Comment:
nit: This case doesn't relate to unwrap, right?

Reply:
It is related. The makerToken of `order[0]` is WETH while takerToken of `order[1]` is ETH. So after collecting WETH from order[0], it has to be unwrapped first before distributing out.